dips/dip-196.md

```diff
@@ -0,0 +1,55 @@
+---
+dip: 196
+title: Onchain Network Identity
+authors: Greg Nazario (@gregnazario)
+status: Draft
+type: Informational
+created: 07/06/2021
+last_updated: 07/06/2021
+issue: https://github.com/diem/dip/issues/196
+---
+
+import useBaseUrl from '@docusaurus/useBaseUrl';
+
+# Summary
+
+Diem nodes need the ability to find other nodes in the network easily.  Additionally, the nodes need the ability to find 
+the associated network identity so that the nodes can trust them.  This DIP describes the ability for node owners to
+self register their network identity onchain, and allow other nodes to lookup this data for discovery of other nodes.
+An example of a use case for this is partners guaranteeing access to Validator Full Nodes.  Validator full nodes can
+easily subscribe and know which nodes to prioritize.
+
+# Specification
+
+## Onchain Network Identity Format
+
+The identities shall be a vector of bcs encoded `NetworkIdentity` objects.  However, the vector shall be a set, and any duplicates will be ignored.
+```
+{
+  identities: vector<vector<u8>>,
+}
+```
+
+The `NetworkIdentity` struct shall be a tuple of `peer_id` and a x25519 `public_key` that represents the identity for a single node.
+```
+{
+    peer_id: <account_address>,
+    public_key: <base64 encoded x25519 public key>,
+}
+```
+
+## Updating Network Identity - Owner
+
+The `NetworkIdentity` for an account shall be updated by only the owner of the account.  At this point, the consumers of this identity must update their local use of identity within a reasonable amount of time.
+
+## Consuming Network Identity
+
+Consumers of `NetworkIdentity` may keep track of a list of accounts to watch.  The consumers then can watch the event streams of each account's `NetworkIdentity` for updates and update accordingly.
+
+# Backwards Compatibility
+
+As this only applies to discovery of future nodes, there are no concerns for backwards compatibility.
+
```

Comment:
Can we add a blurb about applications that tie back into your earlier TBF and connection limitation work?

Reply:
As a way to say how this works together with that work?  Or as a way to justify why VFNs need this identity?